megamek/src/megamek/client/generator/RandomCallsignGenerator.java

```diff
@@ -49,23 +49,53 @@
  * that is used during generation
  */
 public class RandomCallsignGenerator {
-    private final static RandomCallsignGenerator INSTANCE = new RandomCallsignGenerator();
     private final static MMLogger logger = MMLogger.create(RandomCallsignGenerator.class);
 
+    private final static String TEST_DIR = "testresources/data/names/callsigns_test.csv";
+
+    private final static RandomCallsignGenerator INSTANCE = new RandomCallsignGenerator(false);
+    private final static RandomCallsignGenerator TEST_INSTANCE = new RandomCallsignGenerator(true);
+
+
```

Comment:
This is fine, but another approach we can use is extending the main class with a Test<ClassName> class that lives in the unittest code dir.
As long as the namespace (e.g. `client/generator/`) matches, JUnit should find the derived class automatically.
Of course, that approach requires changing the protection level of some internal fields, and for singletons (like this, or Systems) allowing the user to set the instance directly.

Both approaches have benefits and drawbacks.